great_expectations/data_asset/util.py

```diff
@@ -120,7 +121,20 @@ def recursively_convert_to_json_serializable(test_obj):
     except (TypeError, ValueError):
         pass
 
-    if isinstance(test_obj, (str, int, float, bool)):
+    if isinstance(test_obj, str):
+        if "'" in test_obj:
```

Comment:
remark : this function is used in this project not only for `row_condition` 
if some part of the project really needs to save config with single quotes in it, we could find alternative solution

1/ use regex to accept quotes if they is an escape char before ?
this regex should do the trick `if re.search(r"(?<!\\)\'", test_obj):`
but I think in any case we should avoid single in json serialization in any case

2/ find a way to apply this condition only on `row_condition` field. But I don't see where yet

Reply:
Oops there is indeed a problem : 
Example : 
![image](https://user-images.githubusercontent.com/51739671/142275569-0e052d52-1d9d-466f-935f-fb3fa3fb0867.png)
And the regex with escape char doesn't allow me to fix it.
We may need to find another way to solve it.

- fix the way of serialization to allow single quotes ? maybe just a json.dumps() is not well done ?

Reply:
Since the original bug only occurs if the single quote is in `row_condition`, I have find a way to raise the error the prevent the single quotes only for row_condition field. See 3727707